planner/core/explain.go

```diff
@@ -696,8 +700,44 @@ func (p *TiKVSingleGather) ExplainInfo() string {
 
 // ExplainInfo implements Plan interface.
 func (p *PhysicalMemTable) ExplainInfo() string {
-	if !metricschema.IsMetricTable(p.Table.Name.L) {
+	return genMetricTableExplainInfo(p.ctx, p.DBName, p.Table, p.Extractor)
+}
+
+func genMetricTableExplainInfo(ctx sessionctx.Context, dbName model.CIStr, tblInfo *model.TableInfo, extractor MemTablePredicateExtractor) string {
+	if dbName.L != util.MetricSchemaName.L || !metricschema.IsMetricTable(tblInfo.Name.L) {
+		return ""
+	}
+
+	e, ok := extractor.(*MetricTableExtractor)
+	if !ok || e.SkipRequest {
+		return ""
+	}
+	promQL := GetMetricTablePromQL(ctx, tblInfo.Name.L, e.LabelConditions, e.Quantiles)
+	startTime, endTime, step := e.GetQueryRangeTime(ctx)
+	return fmt.Sprintf("PromQL:%v, start_time:%v, end_time:%v, step:%v",
+		promQL,
+		startTime.In(ctx.GetSessionVars().StmtCtx.TimeZone).Format("2006-01-02 15:04:05.999"),
+		endTime.In(ctx.GetSessionVars().StmtCtx.TimeZone).Format("2006-01-02 15:04:05.999"),
+		step,
+	)
+}
+
+// GetMetricTablePromQL uses to get the promQL of metric table.
+func GetMetricTablePromQL(sctx sessionctx.Context, lowerTableName string, labels map[string]set.StringSet, quantiles []float64) string {
+	def, err := metricschema.GetMetricTableDef(lowerTableName)
+	if err != nil {
 		return ""
 	}
-	return metricschema.GetExplainInfo(p.ctx, p.Table.Name.L)
+	if len(quantiles) == 0 {
+		quantiles = []float64{def.Quantile}
+	}
+	var buf bytes.Buffer
+	for i, quantile := range quantiles {
+		promQL := def.GenPromQL(sctx, labels, quantile)
+		if i > 0 {
+			buf.WriteByte('\n')
```

Comment:
Is it proper to use a new line in the explain result?

Reply:
ok, I will change back to `,`